### PR TITLE
[fix] simd concept can't handle incomplete types

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

**Comment by h-2** [comment]
Maybe we should have a generic `CompleteType` concept or at least `is_complete_type_v`?

File: test/unit/core/simd/detail/builtin_simd_test.cpp [hunk 1]
```diff
@@ -72,6 +76,39 @@ using uint32x8_t [[gnu::vector_size(32)]] = uint32_t;
 using uint64x4_t [[gnu::vector_size(32)]] = uint64_t;
 #endif
 
+namespace incomplete
+{
```
Comment (eseiler) [inline test/unit/core/simd/detail/builtin_simd_test.cpp]: new line after opening namespace ?

File: test/unit/core/simd/detail/builtin_simd_test.cpp [hunk 2]
```diff
@@ -72,6 +76,39 @@ using uint32x8_t [[gnu::vector_size(32)]] = uint32_t;
...
+struct type;
+
+template <typename t>
+struct template_type;
```
Comment (eseiler) [inline test/unit/core/simd/detail/builtin_simd_test.cpp]: new line before closing namespace

File: test/unit/core/simd/detail/builtin_simd_test.cpp [hunk 3]
```diff
@@ -72,6 +76,39 @@ using uint32x8_t [[gnu::vector_size(32)]] = uint32_t;
...
+}
```
Comment (eseiler) [inline test/unit/core/simd/detail/builtin_simd_test.cpp]:
```suggestion
} // namespace incomplete
```

File: test/unit/core/simd/detail/builtin_simd_test.cpp [hunk 4]
```diff
@@ -85,10 +122,22 @@ TEST(builtin_simd, builtin_simd)
 
 TEST(builtin_simd, is_builtin_simd)
 {
+
```
Comment (eseiler) [inline test/unit/core/simd/detail/builtin_simd_test.cpp]: remove new line

File: test/unit/core/simd/detail/builtin_simd_test.cpp [hunk 5]
```diff
@@ -45,6 +48,41 @@ using uint32x8_t [[gnu::vector_size(32)]] = uint32_t;
 using uint64x4_t [[gnu::vector_size(32)]] = uint64_t;
 #endif
 
+namespace incomplete
```
Comment (rrahn) [inline test/unit/core/simd/detail/builtin_simd_test.cpp]: you and your namespaces 🤣  Is there any particular reason why this has to be in a namespace?
Why not call the struct `incomplete_template_type`?

File: include/seqan3/core/simd/concept.hpp [hunk 6]
```diff
@@ -67,6 +62,32 @@ SEQAN3_CONCEPT simd_concept = requires (simd_t a, simd_t b)
...
+ * seqan3::simd::simd_concept checks whether a given type is a simd type. One of the prerequisites is
+ * that seqan3::simd::simd_traits is defined for this type.
+ *
+ * \if DEV
+ * \todo Simplify concept to the seqan3::detail::simd_concept once gcc bug is fixed
+ * \endif
+ */
+//!\cond
+template <typename simd_t>
+SEQAN3_CONCEPT simd_concept = !std::is_pointer_v<simd_t> && !std::is_array_v<simd_t> && detail::simd_concept<simd_t>;
```
Comment (rrahn) [inline include/seqan3/core/simd/concept.hpp]:
```suggestion
SEQAN3_CONCEPT simd_concept = !std::is_pointer_v<std::decay_t<simd_t>> && detail::simd_concept<simd_t>;
```

File: test/unit/core/simd/detail/builtin_simd_test.cpp [hunk 7]
```diff
@@ -141,8 +190,19 @@ TEST(builtin_simd, simd_concept)
 {
     EXPECT_FALSE(simd_concept<short>);
     EXPECT_FALSE(simd_concept<int>);
-    EXPECT_FALSE(simd_concept<int[15]>);
-    EXPECT_FALSE(simd_concept<int*>);
+    EXPECT_FALSE(simd_concept<incomplete::type>);
+    EXPECT_FALSE(simd_concept<incomplete::template_type<int>>);
```
Comment (rrahn) [inline test/unit/core/simd/detail/builtin_simd_test.cpp]: so without the explicit protection with the !is_pointer gcc will hard fail on the requires clause? Have you reported this as a bug to gcc? is there any update?

File: test/unit/core/simd/detail/builtin_simd_test.cpp (hunk 5 above)
Comment (marehr) [reply test/unit/core/simd/detail/builtin_simd_test.cpp]: there was a difference between stuff within namespace and stuff without namspace. I test therefore both.

File: test/unit/core/simd/detail/builtin_simd_test.cpp (hunk 7 above)
Comment (marehr) [reply test/unit/core/simd/detail/builtin_simd_test.cpp]: yes, it has no SFINAE behaviour then.

File: include/seqan3/core/simd/concept.hpp (hunk 6 above)
Comment (marehr) [reply include/seqan3/core/simd/concept.hpp]: thank you.

File: test/unit/core/simd/detail/builtin_simd_test.cpp (hunk 5 above)
Comment (rrahn) [reply test/unit/core/simd/detail/builtin_simd_test.cpp]: Can you then please add a comment that this was intended.

**Comment by marehr** [comment]
@rrahn can you rereview?

**Pull request merged**